Refinery 4.0

matiasgarciaisaia opened the pull request:
Hi there!

With @vtiffenberg we've continued the great work of @anitagraham on #479, fixing the specs that were still failing.

We've also replaced the icons usage that was broken on the Refinery upgrade, continuing the work of @l-plan on his [`rails51` branch](https://github.com/l-plan/refinerycms-blog/tree/rails51) - [0ea431b2308874775fd0236b5b47e219e062c30b](https://github.com/l-plan/refinerycms-blog/commit/0ea431b2308874775fd0236b5b47e219e062c30b) specifically.

Hope this is good enough to be merged, so there's official Refinery Blog support for Refinery 4.0 🎉 

Closes #479

bricesanchez commented:
Thanks for your work @matiasgarciaisaia ! I will review it ASAP!

matiasgarciaisaia commented:
Hey, @bricesanchez - would you tag 76ee576e31945dde56ccf45f3715ccf9af631573 as `3-0-stable`?

Refinery 4 isn't compatible with Ruby 2.1 or older, so I'm dropping Ruby 2.0 & 2.1 support from the `.travis.yml` and adding a notice to the `README.md` - and it would be nice to add a link back to the previous version.

I'm not getting all the `ENV['TRAVIS']` conditionals on the `Gemfile`.

https://github.com/refinery/refinerycms-blog/blob/76ee576e31945dde56ccf45f3715ccf9af631573/Gemfile#L22-L37

Do you have [any pointers](https://xkcd.com/138/) on that?

l-plan commented:
hi @matiasgarciaisaia,

I did not expect this fix to be used by anyone, it was jus a quick fix for a project of mine.  Thanks for noticing-but how did you find out anyway? I made some more adjustments, happy to elaborate on them!
grtz, rolf

matiasgarciaisaia commented:
Hi @l-plan!

I've found your repo via the [Network graph](https://github.com/refinery/refinerycms-blog/network) - it was the most updated branch, I think.

I _think_ that most of your other commits are adding more features to the extension (social buttons, sign in links on comments section, etc) - am I right?

This PR is about making the extension work well with Refinery 4. I may review your work later - should I need some of that stuff - but you may want to polish your commits a little bit and create a new PR after this one is merged.

Anyway - thanks for your work! You've really helped ours - we just needed to update the icons, and ended up upgrading the extension 🎉

bricesanchez commented on Gemfile:
```diff
@@ -1,22 +1,23 @@
 end
 
 # Add the default visual editor, for now.
 gem 'refinerycms-wymeditor', ['~> 1.0', '>= 1.0.6']
```
You should use this version of refinerycms-wymeditor : `gem 'refinerycms-wymeditor', ['~> 2.0', '>= 2.0.0']`

bricesanchez commented on config/initializers/assets.rb:
```diff
@@ -0,0 +1 @@
+Rails.application.config.assets.precompile += %w( refinery/blog/backend.js refinery/wymeditor.js )
```
This extension should not be aware of wymeditor in this initializer, you should remove `refinery/wymeditor.js`

bricesanchez replied:
If you have to precompile `refinery/blog/backend` you should add it in https://github.com/refinery/refinerycms-blog/blob/master/lib/refinery/blog/engine.rb#L16-L20

matiasgarciaisaia commented:
(I think I've) Found the issue!

ActiveRecord's PostgreSQL support is fixed to pre-1.0 versions of the `pg` gem - and that's why the postgres builds were failing. I've copied their restrictions, and we will eventually relax it after they merge rails/rails#31671 or something similar.

There was a similar issue in `master` with another dependency - and that's why [the builds were breaking](https://travis-ci.org/refinery/refinerycms-blog/jobs/250891212#L208)

I think I've addressed all of the feedback, and the CI is green. Thanks for all your support! 🎉

matiasgarciaisaia commented:
I'd prefer to redo @anitagraham's commits (split in smaller commits, rewrite the messages, do them step by step) but I don't think I can't afford that much time - sorry about that :/

matiasgarciaisaia commented:
OK, I've at least reworked the commits so there's no longer a merge between them - but the code was kept the same.

Sorry for the repost ¯\_(ツ)_/¯

bricesanchez commented:
Thank you very much @matiasgarciaisaia and @l-plan , i check this soon!

anitagraham commented:
So glad you've fixed the failing specs. I was at an impasse.

bricesanchez commented:
@parndt could you review it?

parndt commented on spec/features/refinery/blog/admin/posts_spec.rb:
```diff
@@ -338,7 +337,7 @@ module Admin
                 click_link "Remove this translation"
 
-                expect(page).not_to have_content(blog_post.title)
+                expect(page).not_to have_content('RU')
```
how come this changed away from the post title?

matiasgarciaisaia replied:
Not sure how to test the i18n feature on Refinery (ie, I start `rails s` at `./spec/dummy` and can't find how to enable i18n), but the spec errors seemed to imply that the index shows something like `EN_POST_TITLE EN RU` for each post, with `EN` & `RU` being links to edit that locale's version of the post.

As this test is removing the RU translation, it won't show `RU` after executing the action - but the post still exists, so that's why you still can find the `blog_post.title` there (you get `EN_POST_TITLE EN`).

matiasgarciaisaia replied:
Not really sure how did the previous version work

parndt replied:
gotcha - that does seem odd that it ever worked 👍

parndt commented:
Thanks @matiasgarciaisaia @vtiffenberg @anitagraham ❤️
